src/GraphQL/parseGraphQLUtils.js

```diff
@@ -12,3 +12,32 @@ export function toGraphQLError(error) {
   }
   return new ApolloError(message, code);
 }
+
+export const extractKeysAndInclude = selectedFields => {
+  selectedFields = selectedFields.filter(
+    field => !field.includes('__typename')
+  );
+  let keys = undefined;
+  let include = undefined;
+  if (selectedFields && selectedFields.length > 0) {
+    keys = [...new Set(selectedFields.map(field => field.split('.')[0]))].join(
```

Comment:
I think I prefer to revert to the optimized way we had and we report the Postgres issue to be solved later (since it is a bug that we should solve anyway). We can mark the test that is failing on Postgres to run only on Mongo. What do you think?